schemes/musica/tuvx/musica_ccpp_tuvx_temperature.F90

```diff
@@ -0,0 +1,70 @@
+module musica_ccpp_tuvx_temperature
+  implicit none
+
+  private
+  public :: create_temperature_profile, set_temperature_values, &
+            temperature_label, temperature_unit
+
+  !> Label for temperature in TUV-x
+  character(len=*), parameter :: temperature_label = "temperature"
+  !> Unit for temperature in TUV-x
+  character(len=*), parameter :: temperature_unit = "K"
+
+contains
+
+  !> Creates a TUV-x temperature profile
+  function create_temperature_profile(height_grid, errmsg, errcode) &
+      result(profile)
+    use musica_ccpp_util,    only: has_error_occurred
+    use musica_tuvx_grid,    only: grid_t
+    use musica_tuvx_profile, only: profile_t
+    use musica_util,         only: error_t
+
+    type(grid_t),     intent(in)  :: height_grid
+    character(len=*), intent(out) :: errmsg
+    integer,          intent(out) :: errcode
+    type(profile_t),  pointer     :: profile
+
+    ! local variables
+    type(error_t) :: error
+
+    profile => profile_t( temperature_label, temperature_unit, &
+                          height_grid, error )
+    if ( has_error_occurred( error, errmsg, errcode ) ) return
+
+  end function create_temperature_profile
+
+  !> Sets TUV-x temperature values from host-model temperatures
+  !!
+  !! See description of `musica_ccpp_tuvx_hegihts_grid.F90` for
+  !! CAM-SIMA <-> TUV-x height grid mapping
+  subroutine set_temperature_values(profile, host_midpoint_temperatures, &
+                                    host_surface_temperature, errmsg, errcode)
+    use musica_ccpp_util,    only: has_error_occurred
+    use musica_tuvx_profile, only: profile_t
+    use musica_util,         only: error_t
+    use ccpp_kinds,          only: kind_phys
+
+    type(profile_t),  intent(inout) :: profile
+    real(kind_phys),  intent(in)    :: host_midpoint_temperatures(:) ! K
+    real(kind_phys),  intent(in)    :: host_surface_temperature      ! K
+    character(len=*), intent(out)   :: errmsg
+    integer,          intent(out)   :: errcode
+
+    ! local variables
+    type(error_t)   :: error
+    real(kind_phys) :: interfaces(size(host_midpoint_temperatures)+2)
+    integer         :: n_host_midpoint_temperatures
+
+    n_host_midpoint_temperatures = size(host_midpoint_temperatures)
+
+    interfaces(1) = host_surface_temperature
```

Comment:
Would it be good to document somewhere (even just in a comment) why `blackbody_temperature_at_surface` is used at the lowest TUV vertical level here instead of the air temperature from the host model elsewhere?

Reply:
I had a similar question regarding what to use for `surface_temperature` according to CCPP standards. Jesse mentioned that we can use `blackbody_temperature_at_surface` for now, assuming that TUV-x needs the temperature everywhere, and this is something that CGD needs to discuss with NOAA. I added [comments](tuvx_run) about this.

Reply:
Just wanted to note here that the reason I suggested the use of `blackbody_temperature_at_surface` is because that is what we currently have as the standard name for `cam_in%ts`, which is what is being used in CAM for this same quantity.  

There is a question that this is maybe similar to `skin_temperature`, which is what NOAA uses in some places, although I believe technically these are different physical quantities.  Either way, I think for now using this standard name will allow the coupling to CAM-SIMA to work appropriately, and we can then always go back and change the standard name itself if we come to an agreement with NOAA.

Reply:
Thank you for providing additional information! I've updated the comments to include it.

Reply:
Thank you!